tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -701,6 +713,18 @@ private void cleanUp() {
     }
   }
 
+  public static TonyClient createClientInstance(String[] args, Configuration conf) throws ParseException {
+    TonyClient client;
+    client = new TonyClient(conf);
+    boolean sanityCheck = client.init(args);
+    client.createYarnClient();
```

Comment:
Can `createYarnClient()` be done as part of `init()` or when `run()` is called on the client? Seems a little weird to call this here.

Also, I think `sanityCheck` should be checked before this line.

Reply:
yah.. this was also some tmp workaround to get the client, gonna clean this up as well.